clients/privacy-center/__tests__/common/geolocation.test.ts

```diff
@@ -178,47 +178,101 @@ describe("getGeolocation", () => {
-      let geolocation = await lookupGeolocation(req);
-      expect(geolocation).toBeNull();
+      try {
+        await lookupGeolocation(req);
+      } catch (error) {
+        if (error instanceof Error) {
+          expect(error.message).toMatch(
+            "Provided location (USA) query parameter is not in ISO 3166 format."
+          );
+        } else {
+          expect(typeof error).toBe(Error);
+        }
+      }
 
       // Test again including a (seemingly valid!) region
       req = createRequest({
         url: "https://privacy.example.com/fides.js?geolocation=USA-NY",
       });
-      geolocation = await lookupGeolocation(req);
-      expect(geolocation).toBeNull();
+      try {
+        await lookupGeolocation(req);
+      } catch (error) {
+        if (error instanceof Error) {
+          expect(error.message).toMatch(
+            "Provided location (USA-NY) query parameter is not in ISO 3166 format."
+          );
+        } else {
+          expect(typeof error).toBe(Error);
+        }
+      }
     });
 
     it("ignores invalid, numeric geolocation query param", async () => {
       const req = createRequest({
         url: "https://privacy.example.com/fides.js?geolocation=12",
       });
-      const geolocation = await lookupGeolocation(req);
-      expect(geolocation).toBeNull();
+      try {
+        await lookupGeolocation(req);
+      } catch (error) {
+        if (error instanceof Error) {
+          expect(error.message).toMatch(
+            "Provided location (12) query parameter is not in ISO 3166 format."
+          );
+        } else {
+          expect(typeof error).toBe(Error);
+        }
+      }
     });
 
     it("ignores invalid, partial locations from geolocation query param (e.g. 'US-')", async () => {
       const req = createRequest({
         url: "https://privacy.example.com/fides.js?geolocation=US-",
       });
-      const geolocation = await lookupGeolocation(req);
-      expect(geolocation).toBeNull();
+      try {
+        await lookupGeolocation(req);
+      } catch (error) {
+        if (error instanceof Error) {
+          expect(error.message).toMatch(
+            "Provided location (US-) query parameter is not in ISO 3166 format."
+          );
+        } else {
+          expect(typeof error).toBe(Error);
+        }
+      }
     });
 
     it("ignores invalid regions from geolocation query param (e.g. 'US-NewYork')", async () => {
       const req = createRequest({
         url: "https://privacy.example.com/fides.js?geolocation=US-NewYork",
       });
-      const geolocation = await lookupGeolocation(req);
-      expect(geolocation).toBeNull();
+      try {
+        await lookupGeolocation(req);
+      } catch (error) {
+        if (error instanceof Error) {
+          expect(error.message).toMatch(
+            "Provided location (US-NewYork) query parameter is not in ISO 3166 format."
+          );
+        } else {
+          expect(typeof error).toBe(Error);
+        }
+      }
```

Comment:
You can simplify these checks using Jest's `toThrow` syntax
```
await expect(lookupGeolocation(req)).rejects.toThrow(
  "Provided location (US-NewYork) query parameter is not in ISO 3166 format."
);
```

Reply:
I think for this one it throws an `Error` object and not a string. So I think this may not work, will check it now and comment back.

Reply:
I was wrong, this works perfectly, I like it, thanks!